libs/sdk-flutter/android/src/main/kotlin/com/breez/breez_sdk/SdkLogInitializer.kt

```diff
@@ -0,0 +1,26 @@
+package com.breez.breez_sdk
+
+import breez_sdk.setLogStream
+
+object SdkLogInitializer {
+    private var nodeLogStream: SdkLogListener? = null
+
+    fun initializeNodeLogStream(): SdkLogListener {
+        if (nodeLogStream == null) {
+            try {
+                nodeLogStream = SdkLogListener()
+                setLogStream(nodeLogStream!!)
+            } catch (e: Throwable) {
+                // Reset nodeLogStream if setting log stream fails
+                e.printStackTrace()
+                nodeLogStream = null
+                throw e
+            }
+        }
+        return nodeLogStream!!
+    }
+
+    fun unsubscribeNodeLogStream() {
+        nodeLogStream?.unsubscribe()
```

Comment:
This will cancel all subscriptions, won't it? We only want to unsubscribe the current one in case there are multiple subscriptions.

Reply:
Yeah, my bad- forgot to pass the CoroutineScope that subscribed to the log events.